### PR TITLE
Replace retry loop with Retry from urllib3

### PR DESCRIPTION
**Pull request by kalikiana**
See: https://progress.opensuse.org/issues/93943

### CODE REVIEW

File: openqa_review/browser.py [hunk 1]
```diff
@@ -112,52 +115,40 @@ def get_page(self, url, as_json=False, cache=True):
         return content
 
     def _get(self, url, as_json=False):  # pragma: no cover
-        last_error = ""
-        for i in range(1, 7):
+        retries = Retry(total=7, backoff_factor=2, status_forcelist=[429, 500, 502, 503, 504])
+        http = requests.Session()
+        http.mount("https://", HTTPAdapter(max_retries=retries))
```
Comment (okurz) [inline openqa_review/browser.py:120]: what does the "https://" mean at this point? Why does it have to be specified? Does it prevent the use of http://?

**Review by okurz — CHANGES_REQUESTED** [review]

File: openqa_review/browser.py (hunk 1 above)
Comment (kalikiana) [reply openqa_review/browser.py:120]: It means we have an adapter for the gven scheme. You could have different ones per scheme depending on requirements, see https://docs.python-requests.org/en/master/user/advanced/#transport-adapters
So yes. I need to add that as well here. Thank you for noticing!

File: openqa_review/browser.py [hunk 2]
```diff
@@ -14,6 +14,9 @@
 import ssl
 
 import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+import urllib3
```
Comment (Martchus) [inline openqa_review/browser.py:19]: Looks like a new dependency. Do we need to add it somewhere?

Comment (perlpunk) [reply openqa_review/browser.py:19]: probably here: https://github.com/os-autoinst/openqa_review/blob/master/setup.py#L33

File: openqa_review/browser.py (hunk 1 above)
Comment (okurz) [reply openqa_review/browser.py:120]: I suggest something like the following. Some lines below there is already `server_name = urlparse(url).netloc` which you can pull up here, for example:

```suggestion
        u = urlparse(url)
        http.mount(u.scheme, HTTPAdapter(max_retries=retries))
        …
        server_name = u.netloc
        
```

**Review by okurz — APPROVED** [review]

**Pull request merged**